### DISCUSSION TITLE
Allow PCSX-Redux users to add conditional breakpoints

### DISCUSSION THREAD

**Discussion by ghost** [grumpycoders/pcsx-redux#1070] [Ideas]
A conditional breakpoint is a breakpoint that is triggered only when a defined condition is true.

For example many debuggers find it useful to add a breakpoint that pauses emulation only when the $v0 register is equal or not equal to some number.

In the Breakpoints window just add a TextBox where the user can define the condition of the breakpoint that he or she want to add.

If the TextBox is empty then the new breakpoint is unconditional.

**Comment by nicolasnoble** [comment]
So, there is already support for conditional breakpoints, and so much more. The UI/UX for it can definitely be improved, but check out https://pcsx-redux.consoledev.net/lua/#crash-bandicoot-using-conditional-breakpoints

Long story short, you can open the Lua console and type the following:

```lua
bp = PCSX.addBreakpoint(0x80012480, 'Exec', 4, 'Some Conditional Breakpoint', function()
    local regs = PCSX.getRegisters()
    if regs.GPR.n.v0 == 0 then PCSX.pauseEmulator() end
end)
```

**Reply by ghost** [reply]
> So, there is already support for conditional breakpoints, and so much more. The UI/UX for it can definitely be improved, but check out https://pcsx-redux.consoledev.net/lua/#crash-bandicoot-using-conditional-breakpoints
> 
> Long story short, you can open the Lua console and type the following:
> 
> ```lua
> bp = PCSX.addBreakpoint(0x80012480, 'Exec', 4, 'Some Conditional Breakpoint', function()
>     local regs = PCSX.getRegisters()
>     if regs.GPR.n.v0 == 0 then PCSX.pauseEmulator() end
> end)
> ```

I didn't know about the Lua console. I am a Windows user and also a GUI user and I like to use GUI where possible so my request is then to add a new TextBox to the Breakpoints window/tab which can be used to add conditional breakpoints through the GUI instead without Lua console then.

This feature request can receive a low priority because the Lua console can already be used meanwhile to add conditional breakpoints as you said.
